advisor/src/main/kotlin/advisors/Osv.kt

```diff
@@ -154,6 +154,7 @@ private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest? {
 
     if (name.isNotBlank() && pkg.id.version.isNotBlank() && !ecosystem.isNullOrBlank()) {
         return VulnerabilitiesForPackageRequest(
+            // Do not specify the purl here as it is mutually exclusive with the ecosystem.
```

Comment:
Consider making this comment also explain why it is preferable to use ecosystem.